GVRf/Framework/framework/src/main/java/org/gearvrf/io/GVRMouseDeviceManager.java

```diff
@@ -75,9 +80,17 @@ void removeCursorController(GVRBaseController controller) {
             thread.quitSafely();
             thread = new EventHandlerThread(THREAD_NAME);
             threadStarted = false;
+            if(gvrContext != null){
+                gvrContext.unregisterDrawFrameListener(this);
+            }
         }
     }
 
+    @Override
+    public void onDrawFrame(float frameTime) {
+        thread.invalidate();
```

Comment:
Do we have per frame callbacks on the framework thread? Any idea on how to subscribe to it?

On the second comment - perhaps invalidate is not a good name. Will change it.